app/controllers/application_controller.rb

```diff
@@ -4,4 +4,21 @@ class ApplicationController < LaaMultiStepForms::ApplicationController
   include CookieConcern
 
   before_action :set_default_cookies
+  before_action :can_access_service
+
+  private
+
+  def can_access_service
+    return unless current_provider
+    return if Providers::Gatekeeper.new(current_provider).provider_enrolled?(service:)
```

Comment:
It makes me a bit nervous that Gatekeeper expects to be initialised with an `auth_info` object, and we're passing in a provider here. It happens to be the case that both objects have an `office_codes` array attribute, but I think it might be worth tweaking the OmniauthCallbacksController so that it sends the provider instance to Gatekeeper, so we always know what sort of thing we're dealing with.

Reply:
We chat separately about this... 

The issue I see (might not be an issue) is that this would create the provider records even if they are not able to use the system....